pkg/gateway/api/api_grpc.go

```diff
@@ -349,6 +349,87 @@ func (s *grpcGatewayService) validateGetEvaluationsRequest(req *gwproto.GetEvalu
 	return nil
 }
 
+/*
+getPrerequisiteDownwards recursively gets the features specified as prerequisite by the targetFeatures.
+*/
+func (s *grpcGatewayService) getPrerequisiteDownwards(
```

Comment:
Let me introduce the way without using a recursive approach.

```suggestion
func (s *grpcGatewayService) getPrerequisiteDownwards(
	targetFeatures, allFeatures map[string]*featureproto.Feature,
) (map[string]*featureproto.Feature, error) {
	prerequisites := make(map[string]*featureproto.Feature, 0)
	// depth first search
	queue := make([]*featureproto.Feature, 0, len(targetFeatures))
	for _, f := range targetFeatures {
		queue = append(queue, f)
	}
	for len(queue) > 0 {
		f := queue[0]
		for _, p := range f.Prerequisites {
			preFeature, ok := allFeatures[p.FeatureId]
			if !ok {
				return nil, ErrFeatureNotFound
			}
			prerequisites[preFeature.Id] = preFeature
			queue = append(queue, preFeature)
		}
		queue = queue[1:]
	}
	if len(prerequisites) == 0 {
		return targetFeatures, nil
	}
	return s.mapMerge(targetFeatures, prerequisites), nil
}

```

Reply:
It's up to you, though.

Reply:
Good!! Thank you 😄